Enyim.Caching/Memcached/MemcachedNode.cs

```diff
@@ -387,6 +394,95 @@ internal async Task InitPoolAsync()
                 }
             }
 
+            private void StartReconciliationTask()
+            {
+                if (_connectionIdleTimeout == TimeSpan.Zero)
+                    return;
+
+                var reconcileTimer = new PeriodicTimer(_connectionIdleTimeout);
+                _ = RunTimer();
+
+                async Task RunTimer()
+                {
+                    while (await reconcileTimer.WaitForNextTickAsync().ConfigureAwait(false))
+                    {
+                        try
+                        {
+                            using var source = new CancellationTokenSource(_connectionIdleTimeout);
+                            await reconcileAsync(source.Token).ConfigureAwait(false);
+                        }
+                        catch(Exception e)
+                        {
+                            _logger.LogWarning("ReconciliationTaskFailed", new EventId(0), e);
+                        }
+                    }
+                }
+
+            }
+
+            private async Task reconcileAsync(CancellationToken cancellationToken)
+            {
+                // synchronize access to this method as only one clean routine should be run at a time
+                await _cleanSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
+                try
+                {
+                    var waitTimeout = TimeSpan.FromMilliseconds(10);
+                    while (true)
+                    {
+                        lock (_freeItems)
+                        {
+                            if (maxItems - _semaphore.CurrentCount + _freeItems.Count <= minItems)
+                                return;
+                        }
+
+
+                        if (!await _semaphore.WaitAsync(waitTimeout, cancellationToken).ConfigureAwait(false))

```

Comment:
Why are we doing this?

Reply:
Semaphores are essential for managing concurrency. Each connection acquires one semaphore. If a semaphore isn't acquired within the timeout period, it indicates that all threads are busy (i.e., every connection is still in use), so in that case, we should simply return